### PR TITLE
[Refactor] Beam Python Healthcare Connectors

### LINKED ISSUE #26877: [Task][Clean-Up]: Create `Healthcare` Directory under Python SDK IOs
### What needs to happen?

To mirror what we have in the Java IOs [1], we should create a new directory under [2], and add the DICOM Python IO. This will make the codebase readability cleaner and give a location for folks to add more healthcare IOs in Python if the need arises.

[1]
https://github.com/apache/beam/tree/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare

[2]
https://github.com/apache/beam/tree/master/sdks/python/apache_beam/io/gcp

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by svetakvsundhar**
Fixes #26877 by

1) Adding a healthcare directory: `apache_beam/io/gcp/healthcare` and migrating Dicom IO there.
2) Leaving warnings in the previous DicomClient and DicomIO locations to prevent breaking changes.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by svetakvsundhar** [comment]
R: @johnjcasey 
R: @tvalentyn

**Comment by johnjcasey** [comment]
Should we have just a warning, or should we reference the new location from the old location to maintain compatability?

**Comment by svetakvsundhar** [comment]
> Should we have just a warning, or should we reference the new location from the old location to maintain compatability?

Just so that IIUC, you are referring to something like that is done in https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/external/kafka.py#L24?

If so, that is probably what we should do

**Comment by svetakvsundhar** [comment]
@tvalentyn friendly bump

File: sdks/python/apache_beam/io/gcp/healthcare/dicomio_integration_test.py [hunk 1]
```diff
@@ -40,9 +40,9 @@
 
 # pylint: disable=wrong-import-order, wrong-import-position
 try:
-  from apache_beam.io.gcp.dicomclient import DicomApiHttpClient
-  from apache_beam.io.gcp.dicomio import DicomSearch
-  from apache_beam.io.gcp.dicomio import UploadToDicomStore
+  from apache_beam.io.gcp.healthcare.dicomclient import DicomApiHttpClient
```
Comment (tvalentyn) [inline sdks/python/apache_beam/io/gcp/healthcare/dicomio_integration_test.py:43]: Have you tested locally that these tests also pass without any changes?

**Review by tvalentyn — APPROVED** [review]

File: sdks/python/apache_beam/io/gcp/healthcare/dicomio_integration_test.py (hunk 1 above)
Comment (svetakvsundhar) [reply sdks/python/apache_beam/io/gcp/healthcare/dicomio_integration_test.py:43]: Yes

Screenshot 2023-06-02 at 9 45 35 AM
Screenshot 2023-06-02 at 9 45 48 AM

**Pull request merged**